lib/dnsimple/client/registrar.rb

```diff
@@ -41,6 +41,27 @@ def register(account_id, domain_name, attributes = {}, options = {})
         Dnsimple::Response.new(response, Struct::Domain.new(response["data"]))
       end
 
+      # Renews a domain.
+      #
+      # @see https://developer.dnsimple.com/v2/registrar/#renew
+      #
+      # @example Renew example.com for 3 years:
+      #   client.registrar.renew(1010, "example.com", period: 3)
+      #
+      # @param  [Fixnum] account_id the account ID
+      # @param  [#to_s] domain_name The domain name to register.
+      # @param  [Hash] attributes
```

Comment:
Yes, my intention is to rename all the actions and have the scenario you described in the first section.

This is what we already have in the Go client as well.
https://github.com/aetrion/dnsimple-go/blob/master/dnsimple/registrar.go